tests/unit/store/data.spec.js

```diff
@@ -0,0 +1,673 @@
+      }
+    })
+  },
+  task: (len = 5) => {
+    return Array.from({ length: len }, (v, i) => {
+      return {
+        id: 't' + i,
+        name: 'Task ' + i,
+        flow_id: 'f' + Math.floor(Math.random() * 5) // Random flow id between f0 and f5
+      }
+    })
+  }
+}
+
+describe('data Vuex Module', () => {
+  const unsetState = () => {
+    return {
+      activeFlow: null,
+      activeProject: null,
+      activeTask: null,
+      flows: null,
+      projects: null,
+      tasks: null
+    }
+  }
+
+  const setState = () => {
+    const flows = generators.flow()
+    const projects = generators.project()
+    const tasks = generators.task()
+
+    const activeTask = tasks[Math.floor(Math.random() * 5)] // Random activeTask
+    const activeFlow = flows.find(f => f.id == activeTask.flow_id) // activeFlow based on activeTask
+    const activeProject = projects.find(p => p.id == activeFlow.project_id) // activeProject based on activeFlow
+
+    return {
+      activeFlow: activeFlow,
+      activeProject: activeProject,
+      activeTask: activeTask,
+      flows: flows,
+      projects: projects,
+      tasks: tasks
+    }
+  }
+
+  describe('State', () => {
+    it('should hold data', () => {
+      const state = data.state
+      expect(state.activeFlow).toBe(null)
+      expect(state.activeProject).toBe(null)
+      expect(state.activeTask).toBe(null)
+
+      expect(state.flows).toBe(null)
+      expect(state.projects).toBe(null)
+      expect(state.tasks).toBe(null)
+    })
+  })
+
+  describe('Getters', () => {
+    describe('while unset', () => {
+      let state, store
+      beforeEach(() => {
+        state = unsetState()
+        store = new Vuex.Store({
+          state: state,
+          getters: data.getters,
+          actions: data.actions,
+          mutations: data.mutations
+        })
+      })
+
+      dataObjects.forEach(d => {
+        const active = 'active' + d.charAt(0).toUpperCase() + d.slice(1)
+        const plural = d + 's'
+
+        test(`${active} getter should return null`, () => {
+          expect(store.getters[active]).toEqual(null)
+        })
+
+        test(`${plural} getter should return null`, () => {
+          expect(store.getters[plural]).toEqual(null)
+        })
+      })
+    })
+
+    describe('while set', () => {
+      let state, store
+      beforeEach(() => {
+        state = setState()
+        store = new Vuex.Store({
+          state: state,
+          getters: data.getters,
+          actions: data.actions,
+          mutations: data.mutations
+        })
+      })
+
+      dataObjects.forEach(d => {
+        const active = 'active' + d.charAt(0).toUpperCase() + d.slice(1)
+        const plural = d + 's'
+        const id = active + 'Id'
+
+        test(`${active} getter should return the state activeFlow object`, () => {
+          expect(store.getters[active]).toEqual(state[active])
+        })
+
+        test(`${plural} getter should return a list of flows`, () => {
```

Comment:
Just flows?  Looks like this is testing flows, projects and tasks?

Reply:
Oops nope! I added the interpolated variables there, I wrote the non-looped test for flows and forgot to update the description.. fixed now! 😄